src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -1518,7 +1521,25 @@ public IHtmlContent GenerateGroupsAndOptions(string optionLabel, IEnumerable<Sel
             IEnumerable<SelectListItem> selectList,
             ICollection<string> currentValues)
         {
-            var listItemBuilder = new HtmlContentBuilder();
+            var itemsList = selectList as IList<SelectListItem>;
+            if (itemsList == null)
+            {
+                itemsList = selectList.ToList();
+            }
+
+            var count = itemsList.Count;
+            if (optionLabel != null)
+            {
+                count++;
+            }
+
+            // Short-circuit work below if there's nothing to add.
+            if (count == 0)
```

Comment:
Is this a super common scenario - to have a select list with no values? Might not be worth the added testing \ maintaining for this scenario.